tools/journal_proto_processor.cpp

```diff
@@ -947,9 +947,15 @@ void JournalProtoProcessor::ProcessInterchangeMessage(
     cs_interface_type_declaration_[descriptor] =
         "internal partial class " + name + " {\n";
   } else {
-    cs_interface_type_declaration_[descriptor] =
-        "[StructLayout(LayoutKind.Sequential)]\ninternal partial struct " +
-        name + " {\n";
+  MessageOptions const& message_options = descriptor->options();
+  std::string const visibility =
+      message_options.GetExtension(journal::serialization::is_public)
```

Comment:
This makes it possible to write `(is_public) = false`, which is something we try to avoid.  Look how things are done for `is_consumed` around line 365.